tests/metrics_publishing.rs

```diff
@@ -577,78 +577,105 @@ struct TestExporter<T> {
     _meter_provider: SdkMeterProvider,
 }
 
+trait AsAny {
+    fn as_any(&self) -> &dyn std::any::Any;
+}
```

Comment:
I think this is fine for now, but basically what you're doing is extracting heterogenous data from an enum to then convert it to a type you think it was. You could have just added methods like `unwrap_sum` and similar and it'd be a bit clearer without the need for these `Any` shenanigans.

But this is rather something that the upstream API should expose I think, so in the meantime I think this is fine.